### PR TITLE
feat: Add subscribe

### PR DESCRIPTION
**Pull request by Nekotoxin**
Done:
1. extend the favorite type from 3 types to 5 types
image
2. add the topic subscribe feature
image
3. change the favor to focus(contain favor and subscribe)
image
4. new /my/topic page
image
5. reply to a topic will automatically add the topic to subscribe list,and when the topic update again, you will get email notification.
Todo:
add the subscribe feature for single reply.

### CODE REVIEW

**Comment by Nekotoxin** [comment]
image

**Comment by Nekotoxin** [comment]
1.
image
change the favoritesType from int to string
2.
improve the translation

File: controllers/favorites.go [hunk 1]
```diff
@@ -26,8 +26,24 @@ import (
...
 func (c *ApiController) AddFavorites() {
 	objectId := c.Input().Get("id")
-	favoritesTypeStr := c.Input().Get("type")
-	favoritesType := util.ParseInt(favoritesTypeStr)
+	favoritesType := c.Input().Get("type")
+	memberId := c.GetSessionUsername()
+	var resp Response
+
+	if object.IsFavoritesExist(favoritesType) == false {
+		resp = Response{Status: "fail", Msg: "param wrong"}
```
Comment (hsluoyz) [inline controllers/favorites.go]: Use `ResponseOk` and `ResponseError` in all possible places.

File: controllers/reply.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ func (c *ApiController) UpdateReply() {
 }
 
 // @Title AddReply
-// @Tag Reply API
+//@Tag Reply API
```
Comment (hsluoyz) [inline controllers/reply.go]: space

File: web/src/main/NewReplyBox.js [hunk 3]
```diff
@@ -151,6 +152,14 @@ class NewReplyBox extends React.Component {
         });
       }
     });
+    FavoritesBackend.addFavorites(this.state.topicId, "subscribe_topic").then((res) => {
```
Comment (hsluoyz) [inline web/src/main/NewReplyBox.js:156]: new line

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**